From 89c36d73379431f1f7fe7354b23bf22a535085a4 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Mon, 10 Mar 2014 16:42:03 +0000
Subject: [PATCH 26/28] fix leak of trip_ptr in read_trip_points

trip_ptr is being leaked in read_trip_points, ensure we
free it before we return

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_zone_surface.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/thd_zone_surface.cpp b/src/thd_zone_surface.cpp
index a2838fd..7f1cbe9 100644
--- a/src/thd_zone_surface.cpp
+++ b/src/thd_zone_surface.cpp
@@ -73,6 +73,7 @@ int cthd_zone_surface::read_trip_points() {
 	if (add) {
 		trip_points.push_back(*trip_ptr);
 	}
+	delete trip_ptr;
 
 	return THD_SUCCESS;
 }
-- 
1.9.0

